File: dart_native/ios/Classes/DartNativePlugin.m
#import "DartNativePlugin.h"

@implementation DartNativePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"dart_native"
            binaryMessenger:[registrar messenger]];
  DartNativePlugin* instance = [[DartNativePlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
#if TARGET_OS_OSX
      result([@"macOS " stringByAppendingString:NSProcessInfo.processInfo.operatingSystemVersionString]);
#elif TARGET_OS_IOS
      result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
#endif
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: dart_native/macos/Classes/DartNativePlugin.swift
import Cocoa
import FlutterMacOS

public class DartNativePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "dart_native", binaryMessenger: registrar.messenger)
    let instance = DartNativePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
